apps/ledger-live-mobile/src/families/evm/EditTransactionFlow/EditTransactionSummary.tsx

```diff
@@ -0,0 +1,342 @@
+>;
+
+type Props = Navigation;
+
+function EditTransactionSummary({ navigation, route }: Props) {
+  const { colors } = useTheme();
+  const { nextNavigation, overrideAmountLabel, transactionRaw, editType } = route.params;
+
+  const { account, parentAccount } = useSelector(accountScreenSelector(route));
+
+  invariant(account, "account is missing");
+  invariant(transactionRaw, "transactionRaw is missing");
+
+  const {
+    transaction,
+    setTransaction,
+    status: txStatus,
+    bridgePending,
+  } = useBridgeTransaction(() => ({
+    transaction: route.params.transaction,
+    account,
+    parentAccount,
+  }));
+
+  const transactionToUpdate = fromTransactionRaw(transactionRaw) as EvmTransaction;
+
+  const status = getEditTransactionStatus({
+    transaction: transaction as EvmTransaction,
+    transactionToUpdate,
+    status: txStatus,
+    editType,
+  });
+
+  invariant(transaction, "transaction is missing");
+
+  const isNFTSend = isNftTransaction(transaction);
+
+  // handle any edit screen changes like fees changes
+  useTransactionChangeFromNavigation(setTransaction);
+
+  const [highFeesOpen, setHighFeesOpen] = useState(false);
+
+  const navigateToNext = useCallback(() => {
+    if (!nextNavigation) {
+      return;
+    }
+    return (
+      // This component is used in a wild bunch of navigators.
+      // nextNavigation is a param which can have too many shapes
+      // Unfortunately for this reason let's keep it untyped for now.
+      (navigation as StackNavigationProp<{ [key: string]: object }>).navigate(nextNavigation, {
+        ...route.params,
+        transaction,
+        status,
+        selectDeviceLink: true,
+      })
+    );
+  }, [navigation, nextNavigation, route.params, transaction, status]);
+
+  const onAcceptFees = useCallback(() => {
+    setHighFeesOpen(false);
+    navigateToNext();
+  }, [navigateToNext]);
+  const onRejectFees = useCallback(() => {
+    setHighFeesOpen(false);
+  }, []);
+
+  const { amount, totalSpent, errors, warnings } = status;
+
+  const firstError = errors[Object.keys(errors)[0]];
+
+  const mainAccount = getMainAccount(account, parentAccount);
+  const currencyOrToken = getAccountCurrency(account);
+
+  const hasNonEmptySubAccounts =
+    account &&
+    account.type === "Account" &&
+    (account.subAccounts || []).some(subAccount => subAccount.balance.gt(0));
+
+  const onBuyEth = useCallback(() => {
+    navigation.navigate(NavigatorName.Exchange, {
+      screen: ScreenName.ExchangeBuy,
+      params: {
+        defaultAccountId: account?.id,
+        defaultCurrencyId: currencyOrToken?.id,
+      },
+    });
+  }, [navigation, account?.id, currencyOrToken?.id]);
+
+  const onContinue = useCallback(() => {
+    if (warnings.feeTooHigh) {
+      setHighFeesOpen(true);
+      return;
+    }
+    navigateToNext();
+  }, [navigateToNext, warnings.feeTooHigh]);
+
+  // FIXME: why is recipient sometimes empty?
```

Comment:
is it still the case here ?

Reply:
No idea, this is a copy of `src/screens/SendFunds/04-Summary.tsx` file (with some changes)
https://github.com/LedgerHQ/ledger-live/blob/70b4603a6fa34453c6fa6dd89fb3ed12a1038366/apps/ledger-live-mobile/src/screens/SendFunds/04-Summary.tsx#L154

Reply:
Alright, it looks like a FIXME that never going to be handled to me ahah
I think that the check `!transaction` can be removed as it's handled by an invariant few lines above

Reply:
Indeed, some checks are unnecessary here

3ef6e05d4ec4ab99dd05bdde7db4597ea234eac3